tests/test_api_v1.py

```diff
@@ -936,8 +936,20 @@ def test_create_request_connection_error(mock_chain, app, auth_env, client, db):
     )
     rv = client.post("/api/v1/requests", json=data, environ_base=auth_env)
 
+    engine = db.create_engine(app.config["SQLALCHEMY_DATABASE_URI"], {})
+    connection = engine.connect()
+    request_state = db.Table("request_state", db.MetaData(), autoload=True, autoload_with=engine)
+    query = db.select([request_state]).where(request_state.columns.request_id == 1)
+
+    state_reasons = []
+    for res in connection.execute(query):
+        state_reasons.append(res[2])
+
+    error = "Failed to schedule the task to the workers. Please try again."
+    assert any(elem == error for elem in state_reasons)
```

Comment:
This test already queries the database for the request, see below
```python
assert Request.query.get(1).state.state_name == "failed"
```
You can use the same approach to test for the state reason

Reply:
Ok, I didn't know what this line is about because I didn't find query in Request class. Now I get where it's coming from. In that case there's something odd with this test or I just don't understand how that works. But this assertion is passing in both cases: when you run db.session.commit() in create_request() and also if you don't run it there. I think this shouldn't be happening.
I tried to check state with:
```assert Request.query.get(1).state.state_reason == "Failed to schedule the task to the workers. Please try again."```
This also does not reflect if db.session.commit() run and always passes.

Reply:
Can confirm that this code is a proper test while `Request.query` is not, very interesting :D
We can keep your version and later investigate why the previous test does not work.